src/GDBDebugSession.ts

```
@@ -1792,6 +1806,65 @@ export class GDBDebugSession extends LoggingDebugSession {
+        // initialize variables array and dereference the frame handle
+        const variables: DebugProtocol.Variable[] = [];
+        const frame = this.frameHandles.get(ref.frameHandle);
+        if (!frame) {
+            return Promise.resolve(variables);
+        }
+        
+        try {
+            if (this.registerMap.size === 0) {
+                ////// Refer to e2s plugin to port request code
+                const result_names = await mi.sendDataListRegisterNames(this.gdb, {});
+                let idx = 0;
+                const registerNames = result_names['register-names'];
+                for (const regs of registerNames){
+                    if(regs !== ''){
+                        this.registerMap.set(regs, idx);
+                        this.registerMapReverse.set(idx, regs);
+                    }
+                    idx++;
+                }      
+            }            
+        }
+        catch {
+
+           throw new Error('Unable to parse response for reg. names ');
+        }
+
+        try {
+            const result_values = await mi.sendDataListRegisterValues(this.gdb, {fmt:' x'});
+            const reg_values = result_values['register-values'];
+            for (const n of reg_values){
+                const id = Object.values(n)[0];
+                const reg = this.registerMapReverse.get(parseInt(id));
+                if (reg) {
+                    const val = n.value;
+                    const res: DebugProtocol.Variable = {
+                        name: reg,
+                        evaluateName: '$' + reg,
+                        value: val,
+                        variablesReference: 0
+                    };
+                    variables.push(res);
+                } else {
+                    throw new Error('Unable to parse response for reg. values');
+                }
+            }
+        }
+        catch (error) {
```

Don't discard the exception information, use it so that the source of the problem can be tracked down later.

---

Sorry but I cannot catch your idea here.
Could you please explain more?

---

Same as above. I think we can remove "try ... catch(error)" here.

---

Good idea.